rclcpp/test/rclcpp/test_parameter_map.cpp

```diff
@@ -83,6 +88,48 @@ make_node_params(rcl_params_t * c_params, size_t node_idx, std::vector<std::stri
   }
 }
 
+void
+make_node_param_descriptors(
+  rcl_params_t * c_params, size_t node_idx,
+  std::vector<std::string> param_names)
+{
+  rcl_allocator_t alloc = c_params->allocator;
+  ASSERT_LT(node_idx, c_params->num_nodes);
+  ASSERT_GT(param_names.size(), 0u);
+
+  rcl_node_params_descriptors_s * c_node_param_descriptors = &(c_params->descriptors[node_idx]);
+  c_node_param_descriptors->num_params = param_names.size();
+
+  // Copy parameter names
+  c_node_param_descriptors->parameter_names = static_cast<char **>(
+    alloc.allocate(sizeof(char *) * param_names.size(), alloc.state));
+  for (size_t p = 0; p < param_names.size(); ++p) {
+    const std::string & param_name = param_names[p];
+    c_node_param_descriptors->parameter_names[p] = static_cast<char *>(alloc.allocate(
+        sizeof(char) * (param_name.size() + 1), alloc.state));
+    std::snprintf(
+      c_node_param_descriptors->parameter_names[p], param_name.size() + 1, "%s",
+      param_name.c_str());
+  }
+  // zero init parameter value
+  c_node_param_descriptors->parameter_descriptors =
+    static_cast<rcl_param_descriptor_t *>(alloc.allocate(
```

Comment:
@bpwilcox nit: using `zero_allocate()` would have the same net effect.

Reply:
I agree but is there any benefit of changing?